gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -174,39 +190,156 @@ private Path resolvePath(String pattern) {
 
```

Comment:
This method is very long. Can you refactor this method and break it into several methods, e.g., `getCopyableFilesForPartition`, `getCopyableFilesForTable`?